addons/project/models/project_project.py

```diff
@@ -893,24 +893,33 @@ def _get_hide_partner(self):
     @api.model
     def _create_analytic_account_from_values(self, values):
         company = self.env['res.company'].browse(values.get('company_id', False))
-        project_plan, _other_plans = self.env['account.analytic.plan']._get_all_plans()
+        project_plan_id = int(self.env['ir.config_parameter'].sudo().get_param('analytic.analytic_plan_projects'))
+
+        if not project_plan_id:
+            project_plan, _other_plans = self.env['account.analytic.plan']._get_all_plans()
+            project_plan_id = project_plan.id
+
         analytic_account = self.env['account.analytic.account'].create({
             'name': values.get('name', _('Unknown Analytic Account')),
             'company_id': company.id,
             'partner_id': values.get('partner_id'),
-            'plan_id': project_plan.id,
+            'plan_id': project_plan_id,
         })
         return analytic_account
 
     def _create_analytic_account(self):
         for project in self:
             company_id = project.company_id.id
-            project_plan, _other_plans = self.env['account.analytic.plan']._get_all_plans()
+            project_plan_id = int(self.env['ir.config_parameter'].sudo().get_param('analytic.analytic_plan_projects'))
+
+            if not project_plan_id:
+                project_plan, _other_plans = self.env['account.analytic.plan']._get_all_plans()
+                project_plan_id = project_plan.id
             analytic_account = self.env['account.analytic.account'].create({
                 'name': project.name,
                 'company_id': company_id,
                 'partner_id': project.partner_id.id,
-                'plan_id': project_plan.id,
+                'plan_id': project_plan_id,
                 'active': True,
             })
             project.write({'analytic_account_id': analytic_account.id})
```

Comment:
it's not related to this commit, but maybe we could deduplicate the code and use the _create_analytic_account_from_values method here since it's mostly the same, eg:

```suggestion
            project.analytic_account_id = self._create_analytic_account_from_values({
                'company_id': project.company_id.id,
                'name': project.name,
                'plan_id': project_plan_id,
            })
```

Reply:
Wonderful idea, thanks. 💪🏼